.github/workflows/site-cd-workflow.yml

```diff
@@ -0,0 +1,117 @@
+name: Site Deployment
+
+on:
+  workflow_call:
+    inputs:
+      working-directory:
+        required: true
+        type: string
+
+permissions:
+  id-token: write
+  contents: read
+
+env:
+  ARM_CLIENT_ID: ${{ secrets.AZURE_CLIENT_ID }}
+  ARM_SUBSCRIPTION_ID: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+  ARM_TENANT_ID: ${{ secrets.AZURE_TENANT_ID }}
+  ARM_USE_OIDC: true
+  TF_VAR_subscription_id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+  TF_VAR_local_admin_user: ${{ secrets.localAdminUser }}
+  TF_VAR_local_admin_password: ${{ secrets.localAdminPassword }}
+  TF_VAR_domain_admin_user: ${{ secrets.domainAdminUser }}
+  TF_VAR_domain_admin_password: ${{ secrets.domainAdminPassword }}
+  TF_VAR_deployment_user_password: ${{ secrets.deploymentUserPassword }}
+  TF_VAR_service_principal_id: ${{ secrets.servicePrincipalId }}
+  TF_VAR_service_principal_secret: ${{ secrets.servicePrincipalSecret }}
+  TF_VAR_rp_service_principal_object_id: ${{ secrets.rpServicePrincipalObjectId }}
+  TF_VAR_vm_admin_password: ${{ secrets.vmAdminPassword }}
+  TF_VAR_domain_join_password: ${{ secrets.domainJoinPassword }}
+
+jobs:
+  terraform:
+    name: ${{ inputs.working-directory }}
+    # runs-on: [windows-latest]
+    runs-on: [self-hosted]
```

Comment:
## Arc IaC Automation generated
Next steps: 
 - If you use GitHub runner, you need to delete this line(runs-on: [self-hosted]), uncomment the line(runs-on: [windows-latest]) and then resolve this comment
 - If you use self-hosted runner, just resolve this comment:)

If you have any problem, please file an [issue](https://github.com/Azure/Edge-infrastructure-quickstart-template/issues/new/choose) or contact arcIaCSupport@microsoft.com